provider/oci/environ.go

```diff
@@ -322,7 +322,7 @@ func (e *Environ) Create(ctx envcontext.ProviderCallContext, params environs.Cre
 
 // AdoptResources implements environs.Environ.
 func (e *Environ) AdoptResources(ctx envcontext.ProviderCallContext, controllerUUID string, fromVersion version.Number) error {
-	return errors.NotImplementedf("AdoptResources")
+	return nil
```

Comment:
Why we do not longer return NotImplemented? Even if this is not necessary for OCI, this is more useful than returning `nil`.

Reply:
That NYI error is breaking migration on the OCI, and unrelated to this PR, and returning `nil` is just a temporary patch to be able to run and test this PR. (I believe the people filed the bug were using maas). Before merging this PR I'll revert this change and file an LP bug. And after that I might go tackle that too. But currently retuning nil is just temporary and again, will be reverted before landing.